tools/PrintDirectoryTool.py

```diff
@@ -13,23 +13,57 @@ class PrintDirToolInput(ToolInput):
         default=False,
         description="If true, the tool will print the files and directories recursively.",
     )
+    extensions: str = ToolField(
+        default=None,
+        description="If provided, the tool will only print files with the specified extensions. It should be a "
+        "comma-separated list of extensions.",
+    )
+    ignore_folder: str = ToolField(
+        default=None,
+        description="If provided, the tool will ignore the specified folder.",
+    )
 
 
 @traceable
-def get_directory_contents(path, recursive=False):
+def get_directory_contents(path, recursive=False, extensions=None, ignore_folder=None):
+    if extensions is not None and extensions != "" and isinstance(extensions, str):
+        extensions = extensions.split(",")
     if not os.path.exists(path):
         return {"error": f"Path does not exist: {path}"}
-    # Inicializar una variable para almacenar los resultados
     result = ""
+    if recursive:
+        for root, dirs, files in os.walk(path):
+            # Filtra las carpetas ocultas eliminándolas de la lista `dirs`, también se eliminan las carpetas que se
+            # encuentren en la lista `ignore_folder`
+            dirs[:] = [
+                d
+                for d in dirs
+                if not d.startswith(".") and not (ignore_folder and d in ignore_folder)
+            ]
 
-    # Listar y acumular los nombres de archivos y directorios
-    for entry in os.listdir(path):
-        full_path = os.path.join(path, entry)
-        result += full_path + "\n"
-
-        # Si es un directorio y la recursividad está habilitada, acumular los resultados de forma recursiva
-        if recursive and os.path.isdir(full_path):
-            result += get_directory_contents(full_path, recursive)
+            for file in files:
+                # Ignora archivos ocultos
+                if file.startswith(".") or (ignore_folder and file in ignore_folder):
+                    continue
+                if extensions:
+                    if file.endswith(tuple(extensions)):
```

Comment:
This is fine, but, endswith is case sensitive ... the file can have uppercase the file extension